tests/datastore/test_tiered_datastore.nim

```diff
@@ -149,6 +152,55 @@ suite "TieredDatastore":
       (await ds1.get(key)).get.isSome
       (await ds1.get(key)).get.get == bytes
 
-  # asyncTest "query":
-  #   check:
-  #     true
+  asyncTest "query":
+    let
+      ds = TieredDatastore.new(ds1, ds2).get
+
+      key1 = Key.init("a/b").get
+      key2 = Key.init("a/b:c").get
+      key3 = Key.init("a/b:c/d").get
+
+      bytes1  = @[1.byte, 2.byte, 3.byte]
+      bytes2  = @[4.byte, 5.byte, 6.byte]
+      bytes3: seq[byte] = @[]
+
+      queryKey1 = Key.init("a/*").get
```

Comment:
I don't think there is such a thing as a `*` glob, in the key semantics. A key already has hierarchy implied and allows `/a` would mean to match all the keys with prefix `a`, a.k.a do a `*` match.

Reply:
The key semantics are pretty well defined, which is what the query interface uses, there are no special semantics outside of the key itself + some filtering facilities. The original python implementation and it's derivations take a very straightforward approach to deriving this semantics, which are - 

(pseudocode)

- Keys are hierarchical, and each level of the key gets more and more specific 
  - `/a` - `select * from <table> where key like '/a'%`
  - `/a/b` - `select * from <table> where key like '/a/b'%`
- You can specify keys with namespaces `/a/[field]:[name]`
  - `/a/name:Bob` - `select * from <table> where key like '/a/name:Bob'%`

They query interface then allows you to add filters. 

Consider a key such as - `/a/b/name:Bob/age:33`, to filter by name and age we would do something like:

- Key.init('/a/b') and Filter.init('name', 'Bob', '==') and Filter.init('age', '30', '=>')
  - This roughly translates into 
    - `select * from <table> where name = 'Bob' and age => 30 (select substr(...) as name, substr(...) as age from <table> where key like '/a/b'%)`